client/galaxy/scripts/components/ToolRecommendation.vue

```diff
@@ -0,0 +1,243 @@
+<template>
+    <div id="tool-recommendation-view">
+        <div v-if="deprecated == false" class="infomessagelarge">
+            You have used {{ getToolId }} tool. For further analysis, you could try using the following/recommended
+            tools. The recommended tools are shown in the decreasing order of their scores predicted using machine
+            learning analysis on workflows. A tool with a higher score (closer to 100%) may fit better as the following
+            tool than a tool with a lower score. Please click on one of the following/recommended tools to open its
+            definition.
+        </div>
+        <div v-if="deprecated == true" class="warningmessagelarge">
+            You have used {{ getToolId }} tool. {{ deprecatedMessage }}
+        </div>
+    </div>
+</template>
+
+<script>
+import * as d3 from "d3";
+import { getAppRoot } from "onload/loadConfig";
+import axios from "axios";
+
+export default {
+    props: {
+        toolId: {
+            type: String,
+            required: true
+        }
+    },
+    data() {
+        return {
+            deprecated: null,
+            deprecatedMessage: ""
+        };
+    },
+    created: function() {
+        this.loadRecommendations();
+    },
+    computed: {
+        getToolId: function() {
```

Comment:
Please use es6 functions `() => {}`.

Reply:
I tried using this format of specifying functions in `computed` and `methods` but the object `this` stopped getting recognised. Other places the ES6 format works